From: Gerald Schaefer <geraldsc@de.ibm.com>
Subject: qeth: change checksumming default for HiperSockets
References: bnc#583296
Patch-mainline: Upstream submission made

Symptom:     Possible data integrity problem on HiperSockets
Problem:     Deactivate inbound checksumming on HiperSockets is a valid but
             dangerous optimization in case the frame is routed from an OSA
             network to an HiperSockets network.
Solution:    To go for sure we change the default to software checksumming.

Acked-by: John Jolly <jjolly@suse.de>
---

 drivers/s390/net/qeth_core_main.c |    2 --
 1 file changed, 2 deletions(-)

--- a/drivers/s390/net/qeth_core_main.c
+++ b/drivers/s390/net/qeth_core_main.c
@@ -1113,8 +1113,6 @@ static int qeth_setup_card(struct qeth_c
 	card->ipato.enabled = 0;
 	card->ipato.invert4 = 0;
 	card->ipato.invert6 = 0;
-	if (card->info.type == QETH_CARD_TYPE_IQD)
-		card->options.checksum_type = NO_CHECKSUMMING;
 	/* init QDIO stuff */
 	qeth_init_qdio_info(card);
 	return 0;
